app/queries/drug_stocks_query.rb

```diff
@@ -3,46 +3,60 @@ class DrugStocksQuery
 
   CACHE_VERSION = 1
 
-  def initialize(facilities:, for_end_of_month:, include_block_report: true)
-    @include_block_report = include_block_report
+  def initialize(facilities:, for_end_of_month:)
     @facilities = Facility.where(id: facilities)
     set_facility_group
     set_blocks
     @for_end_of_month = for_end_of_month
     @protocol = @facility_group.protocol
+    @district = @facility_group.region
     @state = @facility_group.state
     @latest_drug_stocks = DrugStock.latest_for_facilities(@facilities, @for_end_of_month)
   end
 
-  attr_reader :for_end_of_month, :facilities, :blocks
+  attr_reader :for_end_of_month, :facilities, :blocks, :facility_group
 
   def drug_stocks_report
-    Rails.cache.fetch(drug_stocks_cache_key,
-      expires_in: ENV.fetch("ANALYTICS_DASHBOARD_CACHE_TTL"),
-      force: RequestStore.store[:bust_cache]) do
-      {patient_count: total_patients,
-       patient_days: all_patient_days,
-       drugs_in_stock: all_drugs_in_stock,
-       patient_count_by_facility_id: patient_count_by_facility_id,
-       patient_days_by_facility_id: patient_days_by_facility_id,
-       drugs_in_stock_by_facility_id: drugs_in_stock_by_facility_id,
-       patient_count_by_block_id: @include_block_report ? patient_count_by_block_id : nil,
-       patient_days_by_block_id: @include_block_report ? patient_days_by_block_id : nil,
-       drugs_in_stock_by_block_id: @include_block_report ? drugs_in_stock_by_block_id : nil,
-       last_updated_at: Time.now}.compact
-    end
+    {total_patient_count: district_patient_count,
+     total_drugs_in_stock: total_drugs_in_stock,
+     total_patient_days: total_patient_days,
+
+     district_patient_count: district_patient_count,
+     district_patient_days: district_patient_days,
+     district_drugs_in_stock: district_drugs_in_stock,
+
+     facilities_total_patient_count: facilities_total_patient_count,
+     facilities_total_patient_days: facilities_total_patient_days,
+     facilities_total_drugs_in_stock: facilities_total_drugs_in_stock,
+
+     patient_count_by_facility_id: patient_count_by_facility_id,
+     patient_days_by_facility_id: patient_days_by_facility_id,
+     drugs_in_stock_by_facility_id: drugs_in_stock_by_facility_id,
+
+     patient_count_by_block_id: patient_count_by_block_id,
+     patient_days_by_block_id: patient_days_by_block_id,
+     drugs_in_stock_by_block_id: drugs_in_stock_by_block_id,
+     last_updated_at: Time.now}
```

Comment:
This is probably my favorite section. It's like the table of contents for this query object.